lib/api/controllers/securityController.js

```diff
@@ -753,3 +914,145 @@ function formatUserSearchResult(kuzzle, formatter, raw) {
       return users;
     });
 }
+
+
+/**
+ * Creates a new User object in Kuzzle's database layer
+ *
+ * @param {Kuzzle} kuzzle
+ * @param {Request} request
+ * @param {boolean} restricted
+ * @returns {Promise<Object>}
+ */
+function createUser (kuzzle, request, restricted) {
+  assertHasBody(request);
+  assertIdStartsNotUnderscore(request);
+  assertBodyHasAttribute(request, 'content');
+  assertBodyAttributeType(request, 'content', 'object');
+
+  if (request.input.body.credentials) {
+    assertBodyAttributeType(request, 'credentials', 'object');
+  }
+
+  if (restricted) {
+    if (request.input.body.content.profileIds) {
+      throw new BadRequestError('The request must not specify the body attribute "content.profileIds".');
+    }
+  }
+  else {
+    if (!request.input.body.content.profileIds) {
+      throw new BadRequestError('The request must specify a body attribute "content.profileIds".');
+    }
+    if (!Array.isArray(request.input.body.content.profileIds)) {
+      throw new BadRequestError('The request must specify the body attribute "content.profileIds" of type "array".');
+    }
+  }
+
+  const pojoUser = request.input.body.content;
+  pojoUser._id = request.input.resource._id || uuid.v4();
+
+  if (restricted) {
+    pojoUser.profileIds = kuzzle.config.security.restrictedProfileIds;
+  }
+
+  return Bluebird.coroutine(persistUserAndStrategies)(kuzzle, request, pojoUser);
+}
+
+/**
+ * @param {Kuzzle} kuzzle
+ * @param {Request} request
+ * @param {object} pojoUser
+ */
+function* persistUserAndStrategies (kuzzle, request, pojoUser) {
```

Comment:
I'm not sure of this but... Maybe you wanted to call this function `persistUserAndCredentials`?